src/ui/public/courier/fetch/request/request.js

```diff
@@ -15,12 +15,42 @@ export default function AbstractReqProvider(Private, Promise) {
       this.source = source;
       this.defer = defer || Promise.defer();
       this._whenAbortedHandlers = [];
-
       requestQueue.push(this);
     }
 
+    /**
+     *  Called by the loopers to find requests that should be sent to the
+     *  fetch() module. When a module is sent to fetch() it's _fetchRequested flag
+     *  is set, and this consults that flag so requests are not send to fetch()
+     *  multiple times.
+     *
+     *  @return {Boolean}
+     */
     canStart() {
-      return Boolean(!this.stopped && !this.source._fetchDisabled);
+      return !this._fetchRequested && !this.stopped && !this.source._fetchDisabled;
+    }
+
+    /**
+     *  Used to find requests that were previously sent to the fetch() module but
+     *  have not been started yet, so they can be started.
+     *
+     *  @return {Boolean}
+     */
+    isFetchRequestedAndPending() {
+      return !!this._fetchRequested && !this.started;
```

Comment:
Why the need for the double negative?  Is it to avoid the undefined case?  It'll still all work as expected right?  Maybe just initialize this._fetchRequested to false in that case?

Just seems a little strange when the value is essentially a boolean, and the double negative requires an extra extra brain cycle or two to make the conversion. :)